include/zephyr/net/lwm2m.h

```diff
@@ -2106,17 +2124,33 @@ int lwm2m_engine_send(struct lwm2m_ctx *ctx, char const *path_list[], uint8_t pa
 /** 
  * @brief LwM2M SEND operation to given path list
  *
+ * @deprecated Use lwm2m_send_cb() instead.
+ *
  * @param ctx LwM2M context
  * @param path_list LwM2M path struct list
  * @param path_list_size Length of path list. Max size is CONFIG_LWM2M_COMPOSITE_PATH_LIST_SIZE
  * @param confirmation_request True request confirmation for operation.
- * 
+ *
  * @return 0 for success or negative in case of error.
  *
  */
 int lwm2m_send(struct lwm2m_ctx *ctx, const struct lwm2m_obj_path path_list[],
```

Comment:
This should use `__deprecated`  macro before the declaration.

Reply:
@rlubos Do we have to deprecate this?
If someone don't want to use `_cb` version. I think both can live in the API.

Reply:
Well, `confirmation_request` is a deprecated parameter, I'm not really sure we want to maintain an API like this. Isn't it good-enough that `lwm2m_send_cb()` accepts NULL as a callback in case it's not used?

Optionally, I guess we could keep the function, but we'd need to modify it's signature, to drop the unused parameter.